itests/src/test/java/fi/metatavu/edelphi/test/ui/base/panel/QueryTestsBase.java

```diff
@@ -0,0 +1,201 @@
+    nextPage(7);
+    
+    // Multi select
+    
+    waitPresent("input[name='queryPageType'][value='THESIS_MULTI_SELECT']");
+    waitAndClick(".queryMultiselectQuestionContainer .queryMultiselectListItemContainer:nth-child(3) input");
+    waitAndClick(".queryMultiselectQuestionContainer .queryMultiselectListItemContainer:nth-child(4) input");
+    waitAndClick(".queryMultiselectQuestionContainer .queryMultiselectListItemContainer:nth-child(5) input");
+    nextPage(8);
+    
+    // Order
+    
+    waitPresent("input[name='queryPageType'][value='THESIS_ORDER']");
+    waitAndClick(".queryOrderingField .queryOrderingFieldItemContainer:nth-child(2) .queryOrderingFieldItemMoveUpButton");
+    waitAndClick(".queryOrderingField .queryOrderingFieldItemContainer:nth-child(3) .queryOrderingFieldItemMoveDownButton");
+    nextPage(9);
+    
+    // Form
+    
+    waitPresent("input[name='queryPageType'][value='FORM']");
+    waitAndType(".queryFormFieldContainer input[type='text']", "text");
+    nextPage(10);
+
+    // Background form
+
+    waitPresent("input[name='queryNextPageNumber'][value='11']");
+    waitAndClick(".queryFormFieldContainer:nth-of-type(1) div:nth-of-type(1) input[type='radio']");
+    waitAndClick(".queryFormFieldContainer:nth-of-type(2) div:nth-of-type(2) input[type='radio']");
+    waitAndClick(".queryFormFieldContainer:nth-of-type(3) div:nth-of-type(3) input[type='radio']");
+    waitAndClick(".queryFormFieldContainer:nth-of-type(4) div:nth-of-type(4) input[type='radio']");
+    waitAndClick(".queryFormFieldContainer:nth-of-type(5) div:nth-of-type(5) input[type='radio']");
+    nextPage(11);
+    
+    // Collage 2d
+    
+    waitPresent("input[name='queryPageType'][value='COLLAGE_2D']");
+    assertVisible(".queryCollage2DQuestionFlotrContainer");
+    
+    finishQuery();
+  }
+  
+  @Test
+  public void testTextPage() {
+    login(ADMIN_EMAIL);
+    createTestPanel();
+    createTestQuery("test", "basic-text-page");
+    navigate("/test/test-query");
+    waitAndAssertText(".queryPageTitle", "Text page");
+    assertText(".queryTextContainer div p", "Page with text");
+  }
+  
+  private void createTestPanel() {
+    navigate("/");
+    createPanel("test");
+    panelMocker.addCreatedPanel("test");
+  }
+  
+  private void createTestQuery(String panelUrl, String...pages) {
+    Long folderId = folderMocker.findPanelRoot(panelUrl);
+    long queryId = queryMocker.addQuery("test-query", "test-query", folderId, true, "Test Query", 0, true, null, "ACTIVE");
+    long sectionId = queryMocker.addDefaultSection(queryId);
+    
+    for (int i = 0; i < pages.length; i++) {
+      queryMocker.addPage(i, queryId, sectionId, pages[i]);
+    }
+    
+    saveQuery(panelMocker.findPanelByRootFolder(folderId), queryId);
+  }
+  
+  private void saveQuery(long panelId, long queryId) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "saveQuery" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)